src/Middleware/SocialAuthMiddleware.php

```diff
@@ -175,18 +175,15 @@ protected function _getUser($providerName, ServerRequest $request)
 
                 return false;
             }
-        } elseif ($identity->email) {
-            $user = $this->_userModel->find($finder)
-                ->where([
-                    $this->_userModel->aliasField($this->config('fields.email')) => $identity->email
-                ])
-                ->first();
         }
 
         $profile = $this->_getProfileEntity($providerName, $identity, $accessToken, $profile ?: null);
+        if ($profile->isDirty()) {
+            $this->_saveProfile($profile);
```

Comment:
Hmm.. Saving before user entity is got means the profile could get saved without `user_id`. I don't want that, so will have to add safeguards for that.